game-core/src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java

```diff
@@ -576,20 +576,21 @@ protected String canProduce(final Territory to, final Collection<Unit> units, fi
       return canProduce(producers.iterator().next(), to, units, player);
     }
     final Collection<Territory> failingProducers = new ArrayList<>();
-    String error = "";
+    final StringBuilder error = new StringBuilder();
     for (final Territory producer : producers) {
       final String errorP = canProduce(producer, to, units, player);
       if (errorP != null) {
         failingProducers.add(producer);
         // do not include the error for same territory, if water, because users do not want to see this error report for
         // 99.9% of games
         if (!(producer.equals(to) && producer.isWater())) {
-          error += ", " + errorP;
+          error.append(", ").append(errorP);
         }
       }
     }
```

Comment:
This really looks like a use case for streams and Collectors#joining
However it will require some slight refactoring.